### DISCUSSION TITLE
Sub-Forms with TS

### DISCUSSION THREAD

**Discussion by bestickley** [react-hook-form/react-hook-form#9412] [Q&A] [answered]
Hi, let's say I have a main form with `FormValues` type like:
```ts
type MainFormValues = {
  firstName: string;
  lastName: string;
  organization: string;
  department: string;
}
```
And then let's say I have a sub-form with `FormValues` type like:
```ts
type SubFormValues = {
  firstName: string;
  lastName: string;
}
```
How can I pass the `control` exported from `useForm` from my main form for use in my sub-form without getting the error:
```bash
Type 'Control<{ firstName: string; lastName: string; organization: string; department: string; }, any>' is not assignable to type 'Control<{ firstName: string; lastName: string; }, any>'.
  The types of '_options.resolver' are incompatible between these types.
```

**Comment by chamatt** [comment, marked as answer]
I'm currently solving this problem by doing this: https://github.com/react-hook-form/react-hook-form/discussions/9345#discussioncomment-4132235

**Reply by bestickley** [reply]
Excellent work around. Thanks, @chamatt!
